perses/protocols/nonequilibrium_cycling.py

```diff
@@ -0,0 +1,655 @@
+        free_energy, error = pymbar.bar.BAR(forward_work, reverse_work)
+
+        return free_energy
+
+    def get_uncertainty(self, n_bootstraps=1000):
+        """
+        Estimate uncertainty using standard deviation of the distribution of bootstrapped
+        free energy (dg) samples.
+
+        Parameters
+        ----------
+        n_bootstraps
+
+        Returns
+        -------
+        free_energy_uncertainty: float
+            Uncertainty on the free energy estimate in units of kT.
+
+        """
+        import numpy as np
+        import numpy.typing as npt
+
+        forward: npt.NDArray[float] = np.array(self.data["forward_work"])
+        reverse: npt.NDArray[float] = np.array(self.data["reverse_work"])
+
+        all_dgs = self._do_bootstrap(forward, reverse, n_bootstraps=n_bootstraps)
+
+        # TODO: Check if standard deviation is a good uncertainty estimator
+        return np.std(all_dgs)
+
+    def get_rate_of_convergence(self):
+        ...
+
+    # @lru_cache()
+    def _do_bootstrap(self, forward, reverse, n_bootstraps=1000):
+        """
+        Performs bootstrapping from forward and reverse cumulated works.
+
+        Parameters
+        ----------
+        forward: np.ndarray[float]
+            Array of cumulated works for the forward transformation
+        reverse: np.ndarray[float]
+            Array of cumulated works for the reverse transformation
+
+
+        Returns
+        -------
+        free_energies: np.ndarray[Float]
+            List of bootstrapped free energies in units of kT.
+        """
+        import pymbar
+        import numpy as np
+        import numpy.typing as npt
+
+        # Check to make sure forward and reverse work values match in length
+        assert len(forward) == len(reverse), "Forward and reverse work values are not paired"
+
+        all_dgs: npt.NDArray[float] = np.zeros(n_bootstraps)  # initialize dgs array
+
+        traj_size = len(forward)
+        for i in range(n_bootstraps):
+            # Sample trajectory indices with replacement
+            indices = np.random.choice(np.arange(traj_size), size=[traj_size], replace=True)
+            dg, ddg = pymbar.bar.BAR(forward[indices], reverse[indices])
+            all_dgs[i] = dg
+
+        return all_dgs
+
+
+class NonEquilibriumCyclingProtocol(Protocol):
+    """
+    Run RBFE calculations between two chemical states using alchemical NEQ cycling and `gufe` objects.
+
+    Chemical states are assumed to have the same component keys, as in, stateA should be composed
+    of the same type of components as components in stateB.
+    """
+
+    result_cls = NonEquilibriumCyclingProtocolResult
+
+    def __init__(self, settings: ProtocolSettings):
+        super().__init__(settings)
+
+    @classmethod
+    def _default_settings(cls) -> ProtocolSettings:
+        from perses.protocols import settings
+        non_eq_settings = settings.NonEqCyclingSettings()
+        return non_eq_settings
```

Comment:
@dotsdl @ijpulidos I think this is meant to be `gufe.Settings` (the big object that contains the `ProtocolSettings` rather than just the `ProtocolSettings`.  This way, if you're very lazy, you can just do `Protocol(settings=Protocol.default_settings())` rather than `Protocol(settings=gufe.Settings(protocol_settings=Protocol.default_settings()))`

Reply:
Yes, I think that makes sense. Good catch! Thanks.

Reply:
This has been implemented in https://github.com/choderalab/perses/pull/1066/commits/9044b3c268be7f469283f570da9fb2b31046a38b